Clarify options placement

LeeLenaleee opened the pull request:
Add clearification to docs by adding the partially quallified options to the name so its more clear where it needs to be. The full path is at the top of the table. This is because otherwise the tables would become scrollable which made it worse to use again.

move responsive, locale, pixel ratio and interactions from general to configuration since it gets put in the options object so it makes more sence to put it there since its also configuration.

Will fix #8417 and #7367

etimberg commented on docs/docs/configuration/elements.md:
```diff
@@ -51,48 +51,48 @@ If the value is an image, that image is drawn on the canvas using [drawImage](ht
-| `borderSkipped` | `string` | `'start'` | Skipped (excluded) border: `'start'`, `'end'`, `'bottom'`, `'left'`, `'top'` or `'right'`.
-| `borderRadius` | `number`\|`object` | `0` | The bar border radius (in pixels).
-| [`pointStyle`](#point-styles) | `string`\|`Image` | `'circle'` | Style of the point for legend.
+| `options.elements.bar.backgroundColor` | [`Color`](../general/colors.md) | `'rgba(0, 0, 0, 0.1)'` | Bar fill color.
```
Should this be `elements.bar` to match the others?

stockiNail commented:
Personally I don't like it because having partial path is more confusing, in my opinion.
Options definition belongs to a section in the doc and then every section can report the namespaces (full paths), for global and not) where those options should be defined,
Just my 2 cents

LeeLenaleee commented:
Yeah @stockiNail, I was planning on using the full qualified names first but then a lot of tables became unusable. Even with partial some became like it and it felt like an unclear mess so I removed that and added above each table the place in the options object

kurkle commented on docs/docs/axes/styling.mdx:
```diff
@@ -8,7 +8,7 @@ There are a number of options to allow styling an axis. There are settings to co
 ## Grid Line Configuration
 
-The grid line configuration is nested under the scale configuration in the `gridLines` key. It defines options for the grid lines that run perpendicular to the axis.
+The scale label configuration can be adjusted in the `options.scales[scaleId].gridLines` namespace. It defines options for the grid lines that run perpendicular to the axis.
```
Looks like a mistake, should talk about grid line.

kurkle commented on docs/docs/axes/styling.mdx:
```diff
@@ -41,7 +41,7 @@ The scriptable context is described in [Options](../general/options.md#tick) sec
 ## Major Tick Configuration
 
-The majorTick configuration is nested under the ticks configuration in the `major` key. It defines options for the major tick marks that are generated by the axis.
+The scale label configuration can be adjusted in the `options.scales[scaleId].ticks.major` namespace. It defines options for the major tick marks that are generated by the axis.
```
Same kind of mistake, talking about major tick

kurkle commented on docs/docs/configuration/animations.mdx:
```diff
@@ -133,18 +133,20 @@ A custom mode can be used by passing a custom `mode` to [update](../developers/a
 A mode option is defined by the same options of the main [animation configuration](#animation-configuration).
 
 ### Default modes
+The animation mode configuration can be adjusted in the `options.animation` namespace.
```
nit: should have a empty line between heading and content (there are more of these, not commenting on all)

kurkle reviewed (commented):
While this looks good, I think plain `Namespace: the.name.space.for.current.thing` would be more efficient
